xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -848,6 +855,73 @@ bool CProcessorHD::IsBT2020Supported()
   return false;
 }
 
+bool CProcessorHD::IsHDR10PassthroughSupported()
+{
+  ComPtr<ID3D11VideoDevice> pVideoDevice;
+  ComPtr<ID3D11VideoProcessorEnumerator> pEnumerator;
+  ComPtr<ID3D11VideoProcessorEnumerator1> pEnumerator1;
+  ComPtr<ID3D11Device> pD3DDevice = DX::DeviceResources::Get()->GetD3DDevice();
+
+  if (FAILED(pD3DDevice.As(&pVideoDevice)))
+  {
+    CLog::LogF(LOGERROR, "failed to get video device.");
+    return false;
+  }
+
+  D3D11_VIDEO_PROCESSOR_CONTENT_DESC desc = {};
+  desc.InputWidth = 3840;
+  desc.InputHeight = 2160;
+  desc.OutputWidth = 3840;
+  desc.OutputHeight = 2160;
+
+  if (FAILED(pVideoDevice->CreateVideoProcessorEnumerator(&desc, &pEnumerator)))
+  {
+    CLog::LogF(LOGERROR, "failed to create Video Enumerator.");
+    return false;
+  }
+
+  if (FAILED(pEnumerator.As(&pEnumerator1)))
+  {
+    CLog::LogF(LOGWARNING,
+               "ID3D11VideoProcessorEnumerator1 is not available. DXVA will not be used.");
+    return false;
+  }
+
+  const DXGI_FORMAT destFormat = DX::Windowing()->GetBackBuffer().GetFormat();
```

Comment:
Problematic with screen SDR/HDR switch by Kodi and AMD "Use 10 bit for SDR" Auto or off.
Back buffer is still 8 bits when this function executes.

The AMD dxva processor doesn't support PQ in B8G8R8A8_UNORM, as a result there is fallback to Pixel Shaders.
The processor has no issue with topleft and left both available in R10G10B10A2 that will be switched to a little later.

Always using R10G10B10A2 format here should be OK, no? There are guards in GetWeight() before getting here.

Reply:
Done